Getting log defaults to stderr

We get content from `stderr` before `stdout` but recently some tests have started to fail on a dataflow side, This is cause by some environments logging `Picked up JAVA_TOOL_OPTIONS` in stderr when `JAVA_TOOL_OPTIONS` is set:
```
2020-08-21T15:30:47.2695234Z [ERROR] Failures: 
2020-08-21T15:30:47.2696373Z [ERROR]   TaskCommandTests.testGetLog:138 Task execution log = CommandResult [success=true, result=Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
```
This kinda shows flaw in an idea that stderr always shows a real error.

We need to change this logic to be more better.

--- a/spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java
+++ b/spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java
@@ -253,10 +253,13 @@ public class LocalAppDeployer extends AbstractLocalDeployerSupport implements Ap
 			for (AppInstance instance : instances) {
 				String stderr = instance.getStdErr();
 				if (StringUtils.hasText(stderr)) {
+					stringBuilder.append("stderr:\n");
 					stringBuilder.append(stderr);
 				}
-				else {
-					stringBuilder.append(instance.getStdOut());
+				String stdout = instance.getStdOut();
+				if (StringUtils.hasText(stdout)) {
+					stringBuilder.append("stdout:\n");
+					stringBuilder.append(stdout);
 				}
 			}
 		}

--- a/spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncher.java
+++ b/spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncher.java
@@ -179,8 +179,18 @@ public class LocalTaskLauncher extends AbstractLocalDeployerSupport implements T
 	public String getLog(String id) {
 		TaskInstance instance = running.get(id);
 		if (instance != null) {
+			StringBuilder stringBuilder = new StringBuilder();
 			String stderr = instance.getStdErr();
-			return (StringUtils.hasText(stderr)) ? stderr : instance.getStdOut();
+			if (StringUtils.hasText(stderr)) {
+				stringBuilder.append("stderr:\n");
+				stringBuilder.append(stderr);
+			}
+			String stdout = instance.getStdOut();
+			if (StringUtils.hasText(stdout)) {
+				stringBuilder.append("stdout:\n");
+				stringBuilder.append(stdout);
+			}
+			return stringBuilder.toString();
 		}
 		else {
 			return "Log could not be retrieved as the task instance is not running.";

